File: schemas/e-commerce/drizzle/product_tags.ts
// product_tags: Lightweight labels for flexible product categorisation.
// See README.md for full design rationale.
import { pgTable, uuid, text, timestamp } from "drizzle-orm/pg-core";

export const productTags = pgTable("product_tags", {
  id: uuid("id").primaryKey().defaultRandom(),
  name: text("name").notNull(),
  slug: text("slug").unique().notNull(),
  description: text("description"),
  createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
});

File: schemas/e-commerce/spacetimedb/product_tags.rs
// product_tags: Lightweight labels for flexible, user-facing product categorization.
// See README.md for full design rationale.
use spacetimedb::Timestamp;

#[spacetimedb::table(name = product_tags, public)]
pub struct ProductTag {
    #[primary_key]
    pub id: String, // UUID

    pub name: String,

    #[unique]
    pub slug: String,

    pub description: Option<String>,

    pub created_at: Timestamp,
}
